package/kedro_viz/models/graph.py

```diff
@@ -497,6 +515,20 @@ def __post_init__(self, data_node: DataNode):
             with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
                 self.plot = json.load(fs_file)
 
+        if data_node.is_metric_node():
+            if not dataset._exists():
+                return
+            
+
+            load_path = get_filepath_str(dataset._get_load_path(), dataset._protocol)
+            with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
+                self.metrics = json.load(fs_file)    
+            self.plot = create_metrics_plot(
+                pd.DataFrame.from_dict(
+                kedro_data_loader.load_all_versions(self.filepath),
+                orient='index'))

```

Comment:
This is personal preference, so feel free to discard, but because of the nesting of so many methods it's a bit hard to read this line. I'd separate the call to `load_all_versions`, so call that first, assign it to a variable and put that into the `dataframe.from_dict`. So you would get:

```
metrics_data = kedro_data_loader.load_all_versions(self.filepath)
self.plot = create_metrics_plot(pd.DataFrame.from_dict(metrics_data, orient='index'))
```